local-volume/provisioner/pkg/common/common.go

```diff
@@ -14,22 +14,27 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package types
+package common
 
 import (
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/cache"
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/util"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
 )
 
 const (
 	AnnProvisionedBy = "pv.kubernetes.io/provisioned-by"
+	// TODO: is this the correct key?
```

Comment:
Yes, I agree with ur understanding, and internally, we always use `--hostname-override` to have consistent hostname seen by kubernetes.

This is unfortunate but do people already use hostname for node affinity?  If so, then we can just conform to that.